File: project_euler/Problem067/Problem067.js
function Problem067() {
    let fileName = 'triangle.txt'
    let border = 100
    let target = Array(border)
    for (let i = 0; i < border; ++i) {
        target[i] = Array(border)
    }

    const file = require('n-readlines')
    let handle = new file(fileName)
    let line
    let index = 0
    while (line = handle.next()) {
        let elements = line.toString().split(' ')
        for (let [i, element] of elements.entries()) {
            target[index][i] = Number(element)
        }
        ++index
    }

    for (let i = border - 1; i > 0; --i) {
        for (let j = i - 1; j >= 0; --j) {
            target[i - 1][j] += Math.max(target[i][j], target[i][j + 1])
        }
    }
    console.log(target[0][0])
}

module.exports = {
    Problem067 : Problem067
};

File: project_euler/Problem067/Problem067.java
package project_euler;

import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStreamReader;

public class Problem067 {
    public void solution() throws IOException {
        String fileName = "triangle.txt";
        int border = 100;
        int[][] target = new int[border][border];

        try (BufferedReader handle = new BufferedReader(new InputStreamReader(new FileInputStream(fileName)))) {
            String line;
            int index = 0;
            while (null != (line = handle.readLine())) {
                String[] elements = line.split(" ");
                for (int i = elements.length - 1; i >= 0; --i) {
                    target[index][i] = Integer.parseInt(elements[i]);
                }
                ++index;
            }
        }

        for (int i = border - 1; i > 0; --i) {
            for (int j = i - 1; j >= 0; --j) {
                target[i - 1][j] += Math.max(target[i][j], target[i][j + 1]);
            }
        }
        System.out.println(target[0][0]);
    }
}
